pkg/reconciler/instances/ory/k8s/ory_crds.go

```diff
@@ -0,0 +1,143 @@
+package k8s
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	apixv1beta1client "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	k8sRetry "k8s.io/client-go/util/retry"
+)
+
+type OryCRDsHandler interface {
+	FindAndDeleteOryFinalizers(kubeconfigData string, logger *zap.SugaredLogger) error
```

Comment:
Interfaces are actually only needed if you plan to provide multiple implementations / allow others to come up with a custom implementation. Otherwise it's cleaner to leave them away and introduce them later when you have a concrete need.

Reply:
In addition to what @tobiscr wrote, there are also some unexported methods in the interface? That's a bit unusual coding style. Any reason for that?

Reply:
I used the interface to match with other ory component code. Unexported methods are removed.

Reply:
Interfaces are typically used in the ory reconciler code to allow for easier mocking. So like @tobiscr said, if you don't  use that mechanism you could introduce it later as well